composers.rb

```diff
@@ -1,34 +1,77 @@
-beethoven = {
-  name: "Ludwig van Beethoven", year_born: 1770, year_died: 1827
-}
-kant = {
-  name: "Immanuel Kant", year_born: 1724, year_died: 1804
-}
-mozart = {
-  name: "Wolfgang Amadeus Mozart", year_born: 1756, year_died: 1791
-}
-
-
-def age(number_1, number_2) 
-  
-  names = ["mozart", "kant", "beethoven"]
-  puts "Which people's age do you want to compare: Beethoven, Kant, Mozart?"
-  number_1 = gets.chomp.downcase.to_sym
-  number_2= gets.chomp.downcase.to_sym
-    
-    names = [mozart, kant, beethoven]
-    hash_name = {mozart: mozart, kant: kant, beethoven: beethoven}
-    name_1 = hash_name[number_1]
-    name_2 = hash_name[number_2]
-  if names.include?(person.to_s)
-    lifetime =person[:year_died] - person[:year_born]
-  puts lifetime
-  else
-    puts "Please enter a valid name"
+# beethoven = {
+#   name: "Ludwig van Beethoven", year_born: 1770, year_died: 1827
+# }
+# kant = {
+#   name: "Immanuel Kant", year_born: 1724, year_died: 1804
+# }
+# mozart = {
+#   name: "Wolfgang Amadeus Mozart", year_born: 1756, year_died: 1791
+# }
+
+@selected_composers = []
+require_relative './composer_class.rb'
+CSV_DELIMITER = ";"
+
+def analyse_composers
+  composers_csv = File.read("composers.csv")
+  composer_lines = composers_csv.split("\n")
+
+  composer_lines.each do |composer_line|
+    parts = composer_line.split(CSV_DELIMITER)
+    composer = Composer.new(parts[0], parts[1], parts[2], parts[3])
+
   end
 end
+def age
+  puts "Which people's age do you want to compare: Beethoven, Kant, Mozart? You can enter as many as you like, then type run if you want to get the results."
+
+  loop do
+    input = gets.chomp.capitalize
+    if check_name(input.capitalize) || input == "Run"
+      Composer.all.each do |composer|
+        if composer.first_name.include?(input) || composer.last_name.include?(input)
+          @selected_composers.push(composer)
+          Composer.selected
+        elsif input == "Run"
+          Composer.selected.each do |composer|
+             puts "#{composer.last_name} grew #{composer.age} years old." 
+
+          end
+          @sorted_composers = Composer.selected.sort_by { |composer| -composer.age }
```

Comment:
Don‘t use an instance variable, if you only need it in the given context (maybe you have to define it outside of the if statement though)